### PR TITLE
ci: update ci workflows for best practices and standards

### LINKED ISSUE #900: update ci workflows for best practices and standards
we need to update the Github Actions CI pipelines to follow industry and organization best practices. This includes but is not limited to using commit locked versions of third-party actions, using self-hosted runners with advanced security, and adding explicit permissions to each workflow.

### PR DESCRIPTION
**Pull request by rbarkerSL**
**Description**:
Updated the workflows to include self-hosted runners, pinned actions, and other best practices. Updated CODEOWNERS and dependabot

**Related issue(s)**:

Fixes #900 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by rbarkerSL** [comment]
PR checks action is failing e2e tests in the same way as on main.

**Comment by gsstoykov** [comment]
Try merging main into the current branch so the Integration tests run with the latest state of main.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -55,28 +83,17 @@ jobs:
...
-      - name: Fetch Hedera SDK
-        uses: actions/checkout@v3
-        with:
-          path: hedera-sdk-go
-
       - name: Run Hedera SDK Tests
         env:
-          OPERATOR_KEY: "302e020100300506032b65700422042091132178e72057a1d7528025956fe39b0b847f200ab59b2fdd367017f3087137"
-          OPERATOR_ID: "0.0.2"
+          OPERATOR_KEY: ${{ secrets.LOCAL_OPERATOR_KEY }}
```
Comment (0xivanov) [inline .github/workflows/main.yml:88]: If a contributor decides to fork the repository and open PR, the secret variables would most likely not be set and the CI checks will fail

**Review by 0xivanov — APPROVED** [review]
LGTM

**Pull request merged**